### PR TITLE
Bugfix/resolve cyclic dependencies

### PR DESCRIPTION
**Pull request by collijk**
### Background
@richbeales was having some cyclic dependency issues because `utils` imported from `logs` and `logs` imported from `utils`.
This resolves that dependency.

### Changes
- merge in the upstream master branch (most of the noise here).
- remove the `send_chat_message_to_user` function
- replace a bare except clause with a specific error.

Will add comments on relevant changes (sorry about the noise).

### PR Quality Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [ ] I have considered potential risks and mitigations for my changes.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: autogpt/agent/agent.py [hunk 1]
```diff
@@ -84,11 +83,7 @@ def start_interaction_loop(self):
                 logger.typewriter_log(
                     "Continuous Limit Reached: ", Fore.YELLOW, f"{cfg.continuous_limit}"
                 )
-                send_chat_message_to_user(
```
Comment (collijk) [inline autogpt/agent/agent.py:87]: Removed a bunch of these

File: autogpt/utils.py [hunk 2]
```diff
@@ -10,24 +10,12 @@
 # Use readline if available (for clean_input)
 try:
     import readline
-except:
```
Comment (collijk) [inline autogpt/utils.py:13]: Made this a proper except clause

File: autogpt/utils.py [hunk 3]
```diff
@@ -10,24 +10,12 @@
...
+except ImportError:
     pass
 
 from autogpt.config import Config
 
 
-def send_chat_message_to_user(report: str):
```
Comment (collijk) [inline autogpt/utils.py:19]: This function does nothing.  `can_handle_report` is not an attribute of the plugin template so this literally always returns without doing anything.

Comment (Wladastic) [reply autogpt/utils.py:19]: this is a new change to the template, it does indeed work

**Pull request merged**